src/psyclone/psyGen.py

```diff
@@ -3526,8 +3412,13 @@ def __init__(self, call, arg_info, access):
             # Use our namespace manager to create a unique name unless
             # the context and label match in which case return the
             # previous name.
-            self._name = self._name_space_manager.create_name(
-                root_name=self._orig_name, context="AlgArgs", label=self._text)
+            # There are unit-tests where we create Arguments without an
+            # associated call.
+            if self._call:
+                tag = "AlgArgs_" + self._text
+                self._name = self._call.root.symbol_table.name_from_tag(
```

Comment:
Again, remove `name_from_tag`. If all three can be replaced then there are none in `psyGen.py`

Reply:
It won't be easy to replace this instance, I added a TODO for #720 instead